test/unit/presenters/formats/completed_transaction_presenter_test.rb

```diff
@@ -0,0 +1,72 @@
+require 'test_helper'
+
+class CompletedTransactionPresenterTest < ActiveSupport::TestCase
+  include GovukContentSchemaTestHelpers::TestUnit
+
+  def subject
+    Formats::CompletedTransactionPresenter.new(edition)
+  end
+
+  def edition
+    @_edition ||= FactoryGirl.create(
+      :completed_transaction_edition,
+      :published,
+      title: "Whacked all moles",
+      slug: "done/good",
+      panopticon_id: artefact.id,
+    )
+  end
+
+  def artefact
+    @_artefact ||= FactoryGirl.create(:artefact, kind: "completed_transaction")
+  end
+
+  def result
+    subject.render_for_publishing_api
+  end
+
+  should "be valid against schema" do
+    assert_valid_against_schema(result, 'completed_transaction')
+  end
+
+  should "[:schema_name]" do
+    assert_equal 'completed_transaction', result[:schema_name]
+  end
+
+  context "[:details]" do
+    should "[:promotion_choice]" do
```

Comment:
We noticed this, so changed it to test for a promotion choice, and added a new test for when there is no promotion.